main.js

```diff
@@ -0,0 +1,212 @@
+const cardList = ['cut', 'attribution', 'home', 'favorite', 'grade', 'bolt', 'key', 'token', 'balance', 'wb_sunny', 'sports_baseball', 'sports_tennis', 'sports_bar', 'motorcycle', 'bomb', 'chess', 'cloud', 'dark_mode', 'local_fire_department', 'flight', 'public', 'spa', 'attach_money', 'savings', 'domino_mask', 'music_note', 'mic', 'face_3', 'joystick', 'visibility'];
+const board = document.querySelector('.game-board');
+const form = document.getElementById('form');
+const difficulties = document.querySelectorAll("input[name='difficulty']");
+const timer = document.getElementById('timer');
+const refresh = document.getElementById('refresh');
+const replay = document.getElementById('replay');
+const modal = document.querySelector('.modal');
+let moveCount = 0;
+let selectedCards = [];
+let cardMatch, NumberOfCards, difficulty, sec, moves, correctMoves, setTimer;
+
+//-----SHUFFLE FUNCTION
+function shuffleCards(list) {
+    for (let i = 0; i < list.length; i += 1) {
+        let j = Math.floor(Math.random() * list.length);
+
+        let temp = list[i];
+        list[i] = list[j];
+        list[j] = temp;
+    }
+}
+
+
+function populate(num) {
+	cardMatch = [];
+	moveCount = 0;
+	board.innerHTML = '';
+	//shuffle (cardList) and take half the number of cards based on difficulty to get a random icons every game
+	shuffleCards(cardList);
+	let cards = cardList.slice(0, num/2);
+	//duplicate the array values to make pairs and shuffle the new array to get random order of cards
+	cards = cards.concat(cards);
+	shuffleCards(cards);
+	//populate HTML
+	const fragment = document.createDocumentFragment();
+	for (let x = 0; x < num; x += 1) {
+		const cardContainer = document.createElement('div');
+		cardContainer.classList.add('card-container', difficulty);
+		const front = document.createElement('div');
+		const back = document.createElement('div');
```

Comment:
Ai o mică breșă de securitate: Dacă cineva se uită în DevTools, poate să vadă ce carte e în spate.

Eu aș încerca să pun conținutul din back doar atunci când este cartea întoarsă. Îl poți pune imediat după click, astfel încât userul să nu vadă.
Atunci când s-a făcut click pe carte greșită, se șterge conținutul din back când [s-a terminat tranziția](https://developer.mozilla.org/en-US/docs/Web/API/Element/transitionend_event).